src/Core/Containers/VariableSet.hpp

```diff
@@ -739,6 +817,12 @@ auto VariableSet::addVariableType() -> Utils::optional<VariableContainer<T>*> {
 
 template <typename T>
 auto VariableSet::existsVariableType() const -> Utils::optional<VariableContainer<T>*> {
+    if constexpr ( std::is_enum<T>::value ) {
```

Comment:
Enum management depends on the usage of the enum, not on the VariableSet management. I think we do not have to impose some restrictions on the variable stored on VariableSet. We should definitly discuss about enum management as an integrated VariableSet service ...